public/app/components/menuBar/menuBar.js

```diff
@@ -26,7 +26,11 @@ angular.module('kibibitCodeEditor')
   };
 })
 
-.controller('menuBarController', function(ngDialog) {
+.controller('menuBarController', function(
+  SettingsService,
+  ngDialog,
+  deviceDetector) {
+
   var vm = this;
   vm.settings = {
```

Comment:
Why do we need this initial _vm.settings_ if we overwrite it in line 48:
`vm.settings = SettingsService.getSettings();`